examples/chip-tool/commands/clusters/Commands.h

```diff
@@ -10702,6 +10723,88 @@ class ReadOnOffClusterRevision : public ModelCommand
 \*----------------------------------------------------------------------------*/
 
+/*
+ * Command AddOpCert
+ */
+class OperationalCredentialsAddOpCert : public ModelCommand
+{
+public:
+    OperationalCredentialsAddOpCert() : ModelCommand("add-op-cert")
+    {
+        AddArgument("noc", &mNoc);
+        AddArgument("iCACertificate", &mICACertificate);
+        AddArgument("iPKValue", &mIPKValue);
+        AddArgument("caseAdminNode", 0, UINT64_MAX, &mCaseAdminNode);
+        AddArgument("adminVendorId", 0, UINT16_MAX, &mAdminVendorId);
+        ModelCommand::AddArguments();
+    }
+    ~OperationalCredentialsAddOpCert()
+    {
+        delete onSuccessCallback;
+        delete onFailureCallback;
+    }
+
+    CHIP_ERROR SendCommand(ChipDevice * device, uint8_t endpointId) override
+    {
+        ChipLogProgress(chipTool, "Sending cluster (0x003E) command (0x06) on endpoint %" PRIu16, endpointId);
+
+        chip::Controller::OperationalCredentialsCluster cluster;
+        cluster.Associate(device, endpointId);
+        return cluster.AddOpCert(
+            onSuccessCallback->Cancel(), onFailureCallback->Cancel(), chip::ByteSpan(chip::Uint8::from_char(mNoc), strlen(mNoc)),
+            chip::ByteSpan(chip::Uint8::from_char(mICACertificate), strlen(mICACertificate)),
+            chip::ByteSpan(chip::Uint8::from_char(mIPKValue), strlen(mIPKValue)), mCaseAdminNode, mAdminVendorId);
+    }
+
+private:
+    chip::Callback::Callback<OperationalCredentialsClusterOpCertResponseCallback> * onSuccessCallback =
+        new chip::Callback::Callback<OperationalCredentialsClusterOpCertResponseCallback>(
+            OnOperationalCredentialsClusterOpCertResponse, this);
+    chip::Callback::Callback<DefaultFailureCallback> * onFailureCallback =
+        new chip::Callback::Callback<DefaultFailureCallback>(OnDefaultFailureResponse, this);
+    char * mNoc;
+    char * mICACertificate;
+    char * mIPKValue;
+    chip::NodeId mCaseAdminNode;
+    uint16_t mAdminVendorId;
+};
+
+/*
+ * Command OpCSRRequest
+ */
+class OperationalCredentialsOpCSRRequest : public ModelCommand
+{
+public:
+    OperationalCredentialsOpCSRRequest() : ModelCommand("op-csrrequest")
+    {
+        AddArgument("cSRNonce", &mCSRNonce);
+        ModelCommand::AddArguments();
+    }
+    ~OperationalCredentialsOpCSRRequest()
+    {
+        delete onSuccessCallback;
+        delete onFailureCallback;
+    }
+
+    CHIP_ERROR SendCommand(ChipDevice * device, uint8_t endpointId) override
+    {
+        ChipLogProgress(chipTool, "Sending cluster (0x003E) command (0x04) on endpoint %" PRIu16, endpointId);
+
+        chip::Controller::OperationalCredentialsCluster cluster;
+        cluster.Associate(device, endpointId);
+        return cluster.OpCSRRequest(onSuccessCallback->Cancel(), onFailureCallback->Cancel(),
+                                    chip::ByteSpan(chip::Uint8::from_char(mCSRNonce), strlen(mCSRNonce)));
```

Comment:
This will crash: mCSRNonce is not a string, it's a byte string. strlen should not be called on any CSR fields.

Reply:
`chip-tool` lacks support of real `OCTET_STRING`. 
Ideally it should take a string as input specified as hex byte and convert it before sending. At the moment, everything is a string passed from the command line. So `strlen` is called on the content of the command line argument. It will likely not crash here, since `mCSRNonce` is a `char *`, but it will probably not work as you may expect.

Reply:
TODO comment then? or will this stop compiling once we support octed_string and will get fixed as a result?